arch/riscv/Kconfig

```diff
@@ -118,6 +118,62 @@ config GEN_IRQ_VECTOR_TABLE
 config NUM_IRQS
 	int
 
+config HAS_MONITOR_HART
+	bool "Does SOC have monitor hart with different capabilities to application harts"
+	default n
+	help
+	  SOC has monitor hart such as the e51 on MPFS and FU540 which has different
+	  capabilities to the other harts in the SOC. This may be important for SMP
+	  setups where the monitor hart may not be suitable for inclusion in the SMP
+	  operation.
+
+config MONITOR_HART_ID
+	int "Hart id for monitor hart"
+	default 0
+	help
+	  If the SOC has monitor hart such as the e51 on MPFS and FU540 this is
+	  the mhartid value for the monitor hart.
+
+config MONITOR_HART_SUPERVISOR
+	bool "Monitor hart supports supervisor mode"
+	default n
+	help
+	  If the SOC has monitor hart, this flag indicates if it supports supervisor
+	  mode. Many monitor harts such as the e51 on MPFS and FU540 do not support
+	  supervisor mode.
+
+config SUPERVISOR_MODE_CAPABLE
```

Comment:
Is there a convenient way to derive this out of the device tree instead? All of these configurations could be described there very nicely through properties on the CPU nodes (and a chosen node to indicate the monitor hart as needed).

Reply:
Hi @olofj Thanks for taking the time to look at this. we have been investigating your suggestion. In the interim, is there a possibility that we could move forward with this PR?